src/runtime/head.ts

```diff
@@ -10,9 +10,9 @@ export function Head(props: HeadProps) {
   let context: ComponentChildren[];
   try {
     context = useContext(HEAD_CONTEXT);
-  } catch {
+  } catch (err) {
     throw new Error(
-      "<Head> component is not supported in the browser, or during suspense renders.",
+      "<Head> component is not supported in the browser, or during suspense renders: " + err.message,
```

Comment:
Seems like a prime candidate for the new `Error.cause` feature in JS: https://github.com/tc39/proposal-error-cause

```suggestion
      "<Head> component is not supported in the browser, or during suspense renders.",
      { cause: err },
```